### PR TITLE
Track additional ad metadata

### PR DESCRIPTION
**Pull request by matamegger**
### Problem

The current integration does not report sufficient ad metadata to the `CISAdAnalytics`.

### Solution

Report ad metadata for client side ads reported through events on the Bitmovin Player.
As there is no known way of retrieving the Google IMA SDK version at runtime it has to be supplied in the MetadataOverrides if it should be tracked. Should we discover an automatic way to retrieve the current version we can improve on the approach.

### Checklist
- [ ] I added an update to `CHANGELOG.md` file
- [x] I ran all the tests in the project and they succeed

### CODE REVIEW

**Comment by matamegger** [comment]
@stonko1994 I would please need some guidance on the 250 body length violation.
(And maybe also some input on the apparently too short variable name 😄 )

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift [hunk 1]
```diff
@@ -385,26 +385,62 @@ public final class ConvivaAnalytics: NSObject {
...
-        if mediaFileUrl != nil {
-            adInfo[CIS_SSDK_METADATA_STREAM_URL] = mediaFileUrl
+        adInfo["c3.ad.position"] = AdEventUtil.parseAdPosition(
+            event: adStartedEvent,
+            contentDuration: player.duration
+        ).rawValue
+        adInfo[CIS_SSDK_METADATA_DURATION] = adStartedEvent.duration
+        adInfo[CIS_SSDK_METADATA_IS_LIVE] = videoAnalytics.getMetadataInfo()[CIS_SSDK_METADATA_IS_LIVE]
+
+        let ad = adStartedEvent.ad
```
Comment (stonko1994) [inline BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift:415]: Let's exclude the term `ad` from being too short in SwiftLint.

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift [hunk 2]
```diff
@@ -385,26 +385,62 @@ public final class ConvivaAnalytics: NSObject {
...
         )
     }
-}
 
-private extension Ad {
-    var adInfo: [String: Any] {
+    private func buildAdInfo(adStartedEvent: AdStartedEvent) -> [String: Any] {
         var adInfo = [String: Any]()
+
+        adInfo["c3.ad.id"] = "NA"
```
Comment (stonko1994) [inline BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift:392]: Can we extract the "NA" into a constant for the not-applicable use-case? 🤔

**Review by stonko1994 — COMMENTED** [review]
Feel free to just silence the Type Body Length Violation

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift (hunk 2 above)
Comment (matamegger) [reply BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift]: Of course. It's NA is not available here though, I think.

https://github.com/bitmovin/bitmovin-player-ios-analytics-conviva/pull/71/commits/b9e646ca057a5972f0dd6b5d22919216cfa2c225

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift (hunk 1 above)
Comment (matamegger) [reply BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift:418]: Sorry for cramping the linting changes in this PR

https://github.com/bitmovin/bitmovin-player-ios-analytics-conviva/pull/71/commits/6f66f9cb87585dd6475a9f51d19efc1e7218c4db
https://github.com/bitmovin/bitmovin-player-ios-analytics-conviva/pull/71/commits/e792261c06395da323b11f193184a717bea8c3dd

**Comment by matamegger** [comment]
As discussed offline: The project setup makes it hard to configure swiftling properly so that the CI as well as Xcode recognize it and behave in the same manner. 
Thus, we just ignore certain rules in place.

File: BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift [hunk 3]
```diff
@@ -385,26 +388,65 @@ public final class ConvivaAnalytics: NSObject {
...
+
+        adInfo["c3.ad.id"] = notAvailable
+        adInfo["c3.ad.system"] = notAvailable
+        adInfo["c3.ad.mediaFileApiFramework"] = notAvailable
+        adInfo["c3.ad.firstAdSystem"] = notAvailable
+        adInfo["c3.ad.firstAdId"] = notAvailable
+        adInfo["c3.ad.firstCreativeId"] = notAvailable
+
         adInfo["c3.ad.technology"] = "Client Side"
+        if adStartedEvent.clientType == AdSourceType.ima {
```
Comment (stonko1994) [inline BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift:403]: You don't need the enum type in Swift.

```suggestion
        if adStartedEvent.clientType == .ima {
```

File: Example/BitmovinConvivaAnalytics/ViewController.swift [hunk 4]
```diff
@@ -67,6 +67,14 @@ class ViewController: UIViewController {
         metadata.custom = ["custom_tag": "Episode"]
         metadata.additionalStandardTags = ["c3.cm.contentType": "VOD"]
 
+        #if os(tvOS)
+        let imaSdkVersion = "4.12.0"
+        #else
+        let imaSdkVersion = "3.22.1"
+        #endif
```
Comment (stonko1994) [inline Example/BitmovinConvivaAnalytics/ViewController.swift:74]: Nit: our internal code style specifies that compiler directives should be written without indentation.

```suggestion
#if os(tvOS)
        let imaSdkVersion = "4.12.0"
#else
        let imaSdkVersion = "3.22.1"
#endif
```

**Review by stonko1994 — COMMENTED** [review]
Please merge develop again into this branch and update the SwiftLint issues accordingly.

File: Example/BitmovinConvivaAnalytics/ViewController.swift (hunk 4 above)
Comment (matamegger) [reply Example/BitmovinConvivaAnalytics/ViewController.swift]: I see, makes sense 👍

**Review by stonko1994 — APPROVED** [review]

**Pull request merged**